checker/check-api-sunset-changed.go

```diff
@@ -0,0 +1,105 @@
+package checker
+
+import (
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/civil"
+	"github.com/tufin/oasdiff/diff"
+)
+
+func APISunsetChangedCheck(diffReport *diff.Diff, operationsSources *diff.OperationsSourcesMap, config BackwardCompatibilityCheckConfig) []BackwardCompatibilityError {
+	result := make([]BackwardCompatibilityError, 0)
+	if diffReport.PathsDiff == nil {
+		return result
+	}
+
+	for path, pathItem := range diffReport.PathsDiff.Modified {
+		if pathItem.OperationsDiff == nil {
+			continue
+		}
+		for operation, operationDiff := range pathItem.OperationsDiff.Modified {
+			op := pathItem.Revision.Operations()[operation]
+			source := (*operationsSources)[op]
+
+			if !op.Deprecated {
+				continue
+			}
+
+			if operationDiff.ExtensionsDiff != nil && operationDiff.ExtensionsDiff.Deleted != nil {
+				result = append(result, BackwardCompatibilityError{
+					Id:        "sunset-deleted",
+					Level:     ERR,
+					Text:      config.i18n("sunset-deleted"),
+					Operation: operation,
+					Path:      path,
+					Source:    source,
+				})
+			}
+
+			if operationDiff.ExtensionsDiff == nil || operationDiff.ExtensionsDiff.Modified == nil {
+				continue
+			}
+
+			opBase := pathItem.Base.Operations()[operation]
+
+			date, err := diff.GetSunsetDate(op.ExtensionProps)
+			if err != nil {
+				result = append(result, BackwardCompatibilityError{
+					Id:        "api-deprecated-sunset-parse",
+					Level:     ERR,
+					Text:      "api sunset date can't be parsed for deprecated API",
+					Operation: operation,
+					Path:      path,
+					Source:    source,
+				})
+				continue
+			}
+
+			baseDate, err := diff.GetSunsetDate(opBase.ExtensionProps)
+			if err != nil {
+				result = append(result, BackwardCompatibilityError{
+					Id:        "api-deprecated-sunset-parse",
+					Level:     ERR,
+					Text:      "api sunset date can't be parsed for deprecated API",
+					Operation: operation,
+					Path:      path,
+					Source:    (*operationsSources)[opBase],
+				})
+				continue
+			}
+
+			days := date.DaysSince(civil.DateOf(time.Now()))
+			deprecationDays := config.MinSunsetStableDays
+
+			stability, err := getStabilityLevel(op.ExtensionProps)
+			if err != nil {
+				result = append(result, BackwardCompatibilityError{
+					Id:        "parsing-error",
+					Level:     ERR,
+					Text:      fmt.Sprintf("parsing error %s", err.Error()),
+					Operation: operation,
+					Path:      path,
+					Source:    source,
+				})
+				continue
+			}
+			if stability == "beta" {
+				deprecationDays = config.MinSunsetBetaDays
+			}
+
+			if baseDate.After(date) && days < deprecationDays {
```

Comment:
This implies that if a sunset date is changed, it must be later than date-of-change + deprecation days.
Was that your intention?

Reply:
If a sunset date is changed, it must be later than original date or later than date-of-change + deprecation days.
So you could move date in future from the original deprication date for any number of days. But if you make it earlier, the deprecation days should be checked.

Reply:
Do you have any thought why it could be bad?